src/s3/types.rs

```diff
@@ -26,7 +26,7 @@ use xmltree::Element;
 
 #[derive(Clone, Debug, Default)]
 /// Contains information of an item of [list_objects()](crate::s3::client::Client::list_objects) API
-pub struct Item {
+pub struct ListItem {
```

Comment:
`Item` is used generically across all SDKs. `ListItem` looks odd.

Reply:
`Item` is too generic here - it clashes with other common types in rust like for example https://docs.rs/futures-core/0.3.28/futures_core/stream/trait.Stream.html and https://doc.rust-lang.org/nightly/std/iter/index.html#iterator

`ListItem` looks fine in the function signature doc (I think):
![image](https://github.com/minio/minio-rs/assets/200059/66b08c17-5e3b-43e7-9915-05f3d9e44bac)

Most users would interact with only one of our SDKs and that usage/experience should be good - almost no one will notice differences in the SDKs.

Reply:
Maybe `ListedItem` is better?

Reply:
I see the name conflict. May be like minio-go's `ObjectInfo` or just `ListEntry`?

Reply:
Done, renamed to `ListEntry`